Web Quiz Engine (Kotlin)/task/src/engine/WebQuizCompletionRepository.kt

```diff
@@ -0,0 +1,14 @@
+package engine
+
+import engine.entity.QuizCompletion
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.stereotype.Repository
+
+typealias QuizCompletionIdentifier = Int
+
+@Repository
+interface WebQuizCompletionRepository: JpaRepository<QuizCompletion, QuizIdentifier> {
+    fun findAllByUserUserIdAndQuizIdIsNotNullAndCompletedAtIsNotNull(userId: Int, pageable: Pageable): Page<QuizCompletion>
```

Comment:
Good job using the JPA Spring language to build a queried method.
It's a good exercise to learn how to write them, but I do think it's kinda over designed in this case.
If you apply to not_null index to your columns as per my recommendation in the other comments, you will not need to add these conditions to your query; you can just do a search by `userId`.

This would be useful if you had use cases where those columns could be null, but you do not.

Reply:
Understood. Done